src/main/java/util/TickingTimer.java

```diff
@@ -158,4 +158,13 @@ public boolean isStarted() {
         return started;
     }
 
+    /**
+     * Changes the timer's period and restart the timer based on the new period.
+     * @param period : the amount of time before triggering the timer.
```

Comment:
`period` is in seconds? Should be mentioned here in the comment, and possibly in the parameter name (rather than in the method name).